src/pytest_split/algorithms.py

```diff
@@ -18,17 +19,26 @@ class TestGroup(NamedTuple):
 def least_duration(splits: int, items: "List[nodes.Item]", durations: "Dict[str, float]") -> "List[TestGroup]":
     """
     Split tests into groups by runtime.
-    Assigns the test with the largest runtime to the test with the smallest
-    duration sum.
+    It walks the test items, starting with the test with largest duration.
+    It assigns the test with the largest runtime to the group with the smallest duration sum.
+
+    The algorithm sorts the items by their duration. Since the sorting algorithm is stable, ties will be broken by
+    maintaining the original order of items. It is therefore important that the order of items be identical on all nodes
+    that use this plugin. Due to issue #25 this might not always be the case.
```

Comment:
Sorting by duration should make it less sensitive to the test order so long as the durations are unique, no?  Seems to me like sorting by duration instead of not sorting at all would fix #25.  Alternatively, what about sorting by the test name, then the duration to resolve any possible ties?

Reply:
> Sorting by duration should make it less sensitive to the test order

Indeed, sorting makes it less sensitive to this problem. Only the collected items that have the same duration will now need to be collected in the right order. However, ties are likely to happen because missing durations get filled with the same average duration.

> Alternatively, what about sorting by the test name, then the duration to resolve any possible ties?

I think we should discuss this further in that issue, and we can adapt this algorithm again later once we agree on a solution. As I mention in that issue, sorting by name would resolve the issue flagged there, but I think collection of objects would still be problematic.